### PR TITLE
`General`: Simplify LTI configuration for instructors

### PR DESCRIPTION
**Pull request by manuelmanso**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added `@PreAuthorize` and checked the course groups for all new REST Calls (security).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
1. We want to make setting up an LTI course as easy as possible for an instructor. Right now, the information for how to setup an LTI exercise is spread out over all exercises in the course. We want to move it all to one place, and make it more clear for instructors what each fields is used for through the use of tooltips.
2. When creating/editing an online course it's not very clear what all the fields are used for, and how they should be filled in by the instructor.
3. When setting up an LTI exercise on another system (e.g. Moodle), it can be annoying to have to copy a lot of different fields manually.
4. When an LTI user is initialized, they are presented with a checkbox explaining their account was created and a password was created as well. They should save the password or they will have to reset it. It's easy for users to click ok without actually reading the dialog.

### Description
1. Moved the LTI configuration page from the exercise page to the course page, split the fields into different tabs to make it clearer what fields belong together and added tooltips. Add edit button to edit configuration. Added tooltips to help the user understand the meaning of each field. Moved the validation logic in the server from CourseResource to OnlineCourseConfigurationService.
2. Removed all the LTI fields from the course edit/create page, course can only be set as online course or not when editing course, and configuration is created with default values.
3. Added a copy button to make copying to clipboard easier.
4. When an LTI user is initialized, they are presented with a checkbox explaining their account was created and a password was created as well. They should save the password or they will have to reset it. It's easy for users to click ok without actually reading the dialog.

### Steps for Testing

Prerequisites:
- 1 Instructor

Course creation:

1. Log in to Artemis
2. Create/edit a course
3. Verify that the LTI fields are not present anymore
4. Verify that creating an online course creates a default configuration
5. Verify that it works to make a non-online course an online course
5. Verify that it works to an online course a non-online course

LTI configuration page:

1. Log in to Artemis
2. Create an online course
3. Navigate to the LTI Configuration page by clicking the "LtiConfiguration" link in the course page
4. Verify that the tabs work and that all information is shown
6. Verify that the copy buttons work
7. Verify that the LTI configuration page is no longer available for exercises

Edit LTI configuration page:

1. Log in to Artemis
2. Create an online course
3. Navigate to the LTI Configuration page by clicking the "LtiConfiguration" link in the course page
4. Verify that clicking the edit button takes you to a new page
5. Verify that the fields have tooltips and are translated
6. Verify that the fields that are required must be filled in
8. Verify that the user prefix field cannot have special characters
9. Verify that the save and cancel button work

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
| Class/File | Branch | Line |
|------------|-------:|-----:|
| OnlineCourseConfiguration.java | 100% | 95% |
| CourseResource.java | 73% | 93% |
| LtiResource.java | 75% | 95% |
| OnlineCourseConfigurationService.java | 85% | 100% |
| LtiDynamicRegistrationService.java | 88% | 100% |
| course-management.service.ts | 90% | 100% |
| course-lti-configuration.component.ts | 76% | 100% |
| edit-course-lti-configuration.component.ts | 76% | 100% |
| copy-icon.button.component.ts | 100% | 100% |

##### Before
ltiUserInitialization

##### After

ltiUserInitAfter

##### Before

https://user-images.githubusercontent.com/11006274/202268368-9d2589bd-db10-4402-ad43-12bcddbfba3a.mp4

##### After

https://user-images.githubusercontent.com/11006274/202268938-a96506c7-843c-4016-b16f-7e7d7b0baf50.mp4

##### Before

https://user-images.githubusercontent.com/11006274/200122142-6397afe4-a54c-4d00-8f1a-5b7289cc0d1a.mp4

##### After

https://user-images.githubusercontent.com/11006274/202269714-ad4e3889-331e-4c7d-955f-7c223a73d003.mp4

### CODE REVIEW

**Review by evgeny1984 — DISMISSED** [review]
Tested on ts1, works as expected.

**Review by kekskatze — DISMISSED** [review]
Tested on testserver 1. Everything worked well

File: src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java [hunk 1]
```diff
@@ -248,6 +253,15 @@ public ResponseEntity<Course> updateCourse(@PathVariable Long courseId, @Request
             courseUpdate.setCourseIcon(pathString);
         }
 
+        if (courseUpdate.isOnlineCourse() != existingCourse.isOnlineCourse()) {
```
[the remark this answers is not in the document]
Comment (manuelmanso) [reply src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java:256]: Made isOnlineCourse return a primitive boolean, there was no reason not to. Thanks for the help

**Review by evgeny1984 — APPROVED** [review]
Tested on tsa2, works as expected after the requested changes.

**Review by sebastiannikles — DISMISSED** [review]
Looks good to me and works as expected.

Only remark: Maybe also make it possible to edit the LTI settings when editing the course (or link from there somehow)? That's where I expected those settings to be at first

**Review by JohannesStoehr — APPROVED** [review]
Thanks for incorporating my suggestions. Code changes since then look good to me

**Review by sebastiannikles — APPROVED** [review]

**Review by kekskatze — APPROVED** [review]
Retested and everything worked well

**Review by sebastiannikles — APPROVED** [review]

**Comment by evgeny1984** [comment]
Some labels translations are inconsistent in german:

- LTI 1.0 Tab:  LTI Geheimnis but LTI Key. Just use English terms or translate it as well.
- LTI 1.3 Tab: Dynamische Registrierung URL but Deep Linking URL and others too.

I think that the terms are specific for LTI, and therefore they don't need to be translated or find a proper German translation.

Otherwise, everything works as expected.
Well done!

File: src/main/webapp/i18n/de/lti.json [hunk 2]
```diff
@@ -9,35 +12,67 @@
...
+            "authorizationUri": "Authorization URI",
+            "tokenUri": "Token URI",
+            "jwkSetUri": "JWKSet URI",
+            "exercises": "Aufgaben",
             "startExercise": "<strong>Herzlich Willkommen bei Artemis!</strong> Klicke auf den Button <i>Aufgabe starten</i>!",
             "ltiSuccessLoginRequired": "LTI Übung erfolgreich gestartet. Bitte melde dich als Nutzer:in {{user}} an.",
             "lti": "LTI",
-            "configurationForExercise": "LTI Konfiguration der Übung {{ title }}",
+            "configurationForCourse": "LTI Konfiguration der Kurs {{ title }}",
             "ltiKey": "LTI Key",
```
Comment (evgeny1984) [inline src/main/webapp/i18n/de/lti.json:36]: LTI Schlüssel

File: src/main/webapp/i18n/de/lti.json [hunk 3]
```diff
@@ -9,35 +12,67 @@
                     "passwordLabel": "Dein Passwort",
                     "passwordHint": "Merk dir das Passwort gut. Du wirst es benötigen um dich manuell einzuloggen oder für unsere Programmieraufgaben.<br>Wenn du dein Passwort vergessen hast oder ändern möchtest kannst du es auf der Login-Seite zurücksetzen. Du kannst das Passwort auch sofort ändern:",
                     "passwordChange": "Passwort ändern",
+                    "readAndUnderstood": "Ich habe die obigen Informationen gelesen und verstehe, dass ich das oben angezeigte Passwort speichern muss.",
                     "confirm": "Bestätigen"
                 }
             },
+            "version10": "LTI 1.0",
+            "version13": "LTI 1.3",
+            "dynamicRegistrationUrl": "Dynamische Registrierung URL",
```
Comment (manuelmanso) [inline src/main/webapp/i18n/de/lti.json:21]:
```suggestion
            "dynamicRegistrationUrl": "Dynamic Registration URL",
```

File: src/main/webapp/i18n/de/lti.json [hunk 4]
```diff
@@ -9,35 +12,67 @@
...
-            "requestUserEmailTooltip": "Die E-Mail des Benutzers muss gesendet werden, da sie von Artemis benötigt wird, um den Benutzer zu erstellen oder zu finden"
+            "requestUserEmailTooltip": "Die E-Mail des Benutzers muss gesendet werden, da sie von Artemis benötigt wird, um den Benutzer zu erstellen oder zu finden",
+            "edit": {
+                "configurationForCourse": "LTI Konfiguration der Kurs {{ title }} bearbeiten",
+                "userPrefixTooltip": "Präfix für die neuen Benutzer, die für diesen Kurs erstellt werden. Es wird empfohlen, einen Wert zu verwenden, der die Benutzer eindeutig als Teil dieses Kurses identifiziert.",
+                "requireExistingUserTooltip": "Wenn ausgewählt, erfordert das Starten der Übung einen bestehenden Benutzer in Artemis. Wenn abgewählt, kann ein Benutzer beim ersten Start erstellt werden, wenn er noch nicht existiert",
+                "ltiKeyTooltip": "Für LTI 1.0 verwendeter Schlüssel. Sie können diesen Wert frei ändern, es wird jedoch empfohlen, den automatisch generierten Wert zu verwenden.",
```
Comment (manuelmanso) [inline src/main/webapp/i18n/de/lti.json:52]:
```suggestion
                "ltiKeyTooltip": "Für LTI 1.0 verwendeter Key. Sie können diesen Wert frei ändern, es wird jedoch empfohlen, den automatisch generierten Wert zu verwenden.",
```

**Review by evgeny1984 — APPROVED** [review]
Tested on ts2, after small requested changes works as expected.

File: src/main/webapp/i18n/de/lti.json [hunk 5]
```diff
@@ -9,35 +12,67 @@
...
+            "clientId": "Client ID",
+            "authorizationUri": "Authorization URI",
+            "tokenUri": "Token URI",
+            "jwkSetUri": "JWKSet URI",
+            "exercises": "Aufgaben",
             "startExercise": "<strong>Herzlich Willkommen bei Artemis!</strong> Klicke auf den Button <i>Aufgabe starten</i>!",
             "ltiSuccessLoginRequired": "LTI Übung erfolgreich gestartet. Bitte melde dich als Nutzer:in {{user}} an.",
             "lti": "LTI",
-            "configurationForExercise": "LTI Konfiguration der Übung {{ title }}",
+            "configurationForCourse": "LTI Konfiguration der Kurs {{ title }}",
```
Comment (JohannesStoehr) [inline src/main/webapp/i18n/de/lti.json:35]:
```suggestion
            "configurationForCourse": "LTI Konfiguration des Kurses {{ title }}",
```

**Review by sebastiannikles — APPROVED** [review]

**Review by kekskatze — APPROVED** [review]
Retested with changed translations. Everything still works

**Pull request merged**